File: Linear Control Course/Chapter10/Lesson5/Chapter10_Lesson5.java
public class RootLocusCase1 {

    public static void main(String[] args) {
        double Kc = 0.05;  // proportional gain from Case Study 1

        // Characteristic: s^2 + 10 s + 1000 Kc = 0
        double a = 1.0;
        double b = 10.0;
        double c = 1000.0 * Kc;

        double disc = b * b - 4.0 * a * c;

        // Complex roots
        double realPart = -b / (2.0 * a);
        double imagPart = Math.sqrt(Math.abs(disc)) / (2.0 * a);

        System.out.println("Closed-loop poles for Kc = " + Kc + ":");
        System.out.println("s1 = " + realPart + " + j" + imagPart);
        System.out.println("s2 = " + realPart + " - j" + imagPart);

        // In a Java-based robot simulator, these poles can be used to set
        // integration step sizes or to compare candidate controller designs
        // before deployment to embedded hardware.
    }
}

File: Linear Control Course/Chapter10/Lesson5/Chapter10_Lesson5.cpp
#include <iostream>
#include <complex>
// Optionally, for larger robotics systems:
// #include <Eigen/Dense>

int main() {
    // Characteristic polynomial: s^2 + 10 s + 1000 Kc = 0
    double Kc = 0.05;  // Case Study 1
    double a = 1.0;
    double b = 10.0;
    double c = 1000.0 * Kc;

    std::complex<double> disc = std::sqrt(
        std::complex<double>(b * b - 4.0 * a * c, 0.0)
    );
    std::complex<double> s1 = (-b + disc) / (2.0 * a);
    std::complex<double> s2 = (-b - disc) / (2.0 * a);

    std::cout << "Closed-loop poles for Kc = " << Kc << ":\n";
    std::cout << "s1 = " << s1 << "\n";
    std::cout << "s2 = " << s2 << "\n";

    // In a real robot joint controller, Kc (and possibly the lead compensator)
    // would be used inside a discrete-time control loop, e.g.
    // u[k] = Kc * (r[k] - y[k]);
    // The plant dynamics would be integrated numerically with a sampling time Ts.
    return 0;
}
